### ISSUE TITLE
after updating tx to latest version i now get an error message "not found"

### ISSUE THREAD

**Issue by Sopor** [transifex/transifex-client#263]
I have been using this command:
`/usr/local/bin/tx -r /home/user/transifex pull -a --mode=reviewed`
to run `tx` for several years now and after i updated to 0.13.6 (today) i now get this error message:
`tx ERROR: Command /home/user/transifex not found`

Doesn't the `-r` working anymore or what is going on?

If i run `tx` from inside the same folder (in this case transifex) it seems to work but i need to use `-r` as i always have done.

**Comment by Sopor** [comment]
I did a test in Windows and it is the exact same behavior as in Linux.

```
D:\tx.exe --version
0.12.1

D:\tx.exe -r D:\transifex_files pull -a --mode=reviewed
New translations found for the following languages:ro_RO
Pulling translations for resource airdcpp.en_installerpot (source: None)
Skipping 'ru_RU' translation (file: ru-RU.po).
Skipping 'no_NO' translation (file: no-NO.po).
Skipping 'fr_FR' translation (file: fr-FR.po).
Skipping 'pt_BR' translation (file: pt-BR.po).
Skipping 'fi_FI' translation (file: fi-FI.po).
Skipping 'pt_PT' translation (file: pt-PT.po).
Skipping 'sv_SE' translation (file: sv-SE.po).
Skipping 'hu_HU' translation (file: hu-HU.po).
Skipping 'nl_NL' translation (file: nl-NL.po).
Skipping 'de_DE' translation (file: de-DE.po).
Skipping 'es_ES' translation (file: es-ES.po).
Pulling new translations for resource airdcpp.en_installerpot (source: None)
Skipping language ro_RO due to used options.
Done.

D:\tx.py36.x64.exe --version
0.13.6, py 3.6, AMD64

D:\tx.py36.x64.exe -r D:\transifex_files pull -a --mode=reviewed
tx ERROR: Command D:\transifex_files not found

```

**Comment by Sopor** [comment]
If i use version 0.13.0 i get this:
```
Traceback (most recent call last):
  File "txclib\cmdline.py", line 100, in <module>
  File "txclib\cmdline.py", line 51, in main
  File "txclib\parsers.py", line 24, in tx_main_parser
TypeError: __init__() got an unexpected keyword argument 'version'
[19740] Failed to execute script cmdline
```

**Comment by Sopor** [comment]
I have reported this already in #148 and i thought it was fixed because i only tested it to run from inside the folder without the switch `-r`, but it seems that it is still not working as it should when using `-r` with full path.

**Comment by Sopor** [comment]
Should it really take 6 month without any response at all? You must have changed something because it is not working as you can see in the above comments but you seems to totally ignore it :(

**Comment by diegobz** [comment]
@Sopor `-r` is an option to specify the name of the resource in Transifex by combining the project and resource slugs. It should look like this `project_slug.resource.slug`.

https://docs.transifex.com/client/pull#pulling-specific-sets-of-translation-files

It seems you are passing a path to that argument.

Look at your `.tx/config` file and check for the names that appear in between brackets. For example, `myproject.main-app` is the name of the resource in the config below.
```
[myproject.main-app]
file_filter = locale/<lang>.po
source_file = locale/en.po
source_lang = en
type = PO
```

We definitely can improve on the error message here though. We will look into it.

**Comment by Sopor** [comment]
@diegobz Why did you close it when it is not working as it should? I have only updated tx and now it suddenly is not working anymore. For some reason the `-r` is not working anymore.
If i go back to version `0.12.1` everything is working again.
```
-r ROOT_DIR, --root=ROOT_DIR
                        change root directory (default is cwd)
```

**Comment by Sopor** [comment]
It seems that you have removed the `-r` so i need to use `--root` instead. Why have you done that?

**Comment by Sopor** [comment]
tx 0.12.1
```
Options:
  --version             show program's version number and exit
  -h, --help            show this help message and exit
  -d, --debug           enable debug messages
  -q, --quiet           don't print status messages to stdout
  -r ROOT_DIR, --root=ROOT_DIR
                        change root directory (default is cwd)
  --traceback           print full traceback on exceptions
  --disable-colors      disable colors in the output of commands
```
tx 0.13.6
```
optional arguments:
  --version         show program's version number and exit
  -d, --debug       enable debug messages
  -q, --quiet       don't print status messages to stdout
  --root ROOT_DIR   change root directory (default is cwd)
  --traceback       print full traceback on exceptions
  --disable-colors  disable colors in the output of commands
```
So they must have been moved the `-r` from `--root` to `--resource`

**Comment by diegobz** [comment]
@Sopor now I see what happened. 

tl;dr: The `-r` option had a double meaning in 0.12.x and we decided to keep it just for --resource on 0.13.*. 

```
$ tx --version
0.12.1
$ tx -h
Usage: tx [options] command [cmd_options]

This is the Transifex command line client which allows you to manage your
translations locally and sync them with the master Transifex server. If you'd
like to check the available commands issue `tx help` or if you just want help
with a specific command issue `tx help command`

Options:
  --version             show program's version number and exit
  -h, --help            show this help message and exit
  -d, --debug           enable debug messages
  -q, --quiet           don't print status messages to stdout
  -r ROOT_DIR, --root=ROOT_DIR
                        change root directory (default is cwd)
  --traceback           print full traceback on exceptions
  --disable-colors      disable colors in the output of commands
```

```
$ tx --version
0.12.1
$tx push -h
Usage: tx [tx_options] push [options]

This command pushes all local files that have been added to Transifex to the
remote server. All new translations are merged with existing ones and if a
language doesn't exists then it gets created. If you want to push the source
file as well (either because this is your first time running the client or
because you just have updated with new entries), use the -f|--force option. By
default, this command will push all files which are watched by Transifex but
you can filter this per resource or/and language.

Options:
  -h, --help            show this help message and exit
  -l LANGUAGES, --language=LANGUAGES
                        Specify which translations you want to push (defaults
                        to all)
  -r RESOURCES, --resource=RESOURCES
                        Specify the resource for which you want to push the
                        translations (defaults to all)
  -f, --force           Push source files without checking modification times.
  --skip                Don't stop on errors. Useful when pushing many files
                        concurrently.
  -s, --source          Push the source file to the server.
  -t, --translations    Push the translation files to the server
  --no-interactive      Don't require user input when forcing a push.
```

We thought that `-r` was never applied to `--root`, because our tests only tested adding the `-r` option after the `pull` or `push` command, like: `tx push -r`. **So we concluded (mistakenly) this wasn't a backward-incompatible change.**

We missed the case where `-r` could be used in front of the command, like `tx -r push`, which in that case it's interpreted as `--root`.

I'm sorry you came across it. 

**The solution is to use `--root` for your use case for versions `>=0.13.0` as you figured out. 👍**
